src/patternfly/components/Table/table.scss

```diff
@@ -1305,7 +1305,6 @@
   // Thead
   thead:where(.#{$table}__thead) {
     th:where(.#{$table}__th),
-    td:where(.#{$table}__td),
```

Comment:
Works for me! I'll just mention that this looks to be due to the checkbox cell specifically since the checkbox has a different `font-size` than the rest of the cells (it's font size * line height makes the cell taller than other 1-line `<th>` cells). So this fixes it, but I dunno if we ever want to put something else in a `thead > td` and have that cell have the same padding as the rest of the `th` cells.

Another approach could be to target something about `.pf-c-table__check` in a compact table > `thead`, or just in a compact table in general. It looks like the presence of the table check element is making all compact table rows taller than they would be if the check wasn't there or used the same font-size as the compact table cell text. Here are 2 duplicate compact tables - one with checkboxes and one without (cc @andrew-ronaldson @mceledonia)

<img width="897" alt="Screenshot 2023-05-17 at 9 18 06 PM" src="https://github.com/patternfly/patternfly/assets/35148959/1bcdc869-960f-4d39-8a8c-9f19c6b04b75">